### PR TITLE
Custom logger

### PR DESCRIPTION
**Pull request by Mannydheer**

### CODE REVIEW

File: server/config/logger.js [hunk 1]
```diff
@@ -1,8 +1,36 @@
...
+const getDateFormat = () => {
+  let date = new Date().toDateString();
+  let time = new Date().toLocaleTimeString();
+  return `${date} || ${time}`;
+};
 
+class Logger {
+  constructor() {
+    //The data relevant about the log.
+    this.logData = null;
```
Comment (quinnlangille) [inline server/config/logger.js:12]: Not sure what this is doing?

Comment (Mannydheer) [reply server/config/logger.js:23]: it is a method defined that will be showing the any data relevant from where I choose to log.

File: server/config/logger.js [hunk 2]
```diff
@@ -1,8 +1,36 @@
...
-class Logger {
-  constructor() {}
+const getDateFormat = () => {
+  let date = new Date().toDateString();
+  let time = new Date().toLocaleTimeString();
+  return `${date} || ${time}`;
+};
 
+class Logger {
+  constructor() {
```
Comment (quinnlangille) [inline server/config/logger.js:21]: To better lock down our singleton instance, `if Logger.instance` is not null, we should throw an error at the start of the contructor

File: server/index.js [hunk 3]
```diff
@@ -70,6 +70,11 @@ const upload = multer({ dest: "./public/uploads/" });
 const PORT = 4000;
 const dbName = "ParkGames";
 
+// LOGGER.
+
+let logger = new Logger(process.env.LOG_LEVEL);
```
Comment (quinnlangille) [inline server/index.js:75]: We wanna call this before the imports above ^, if they call `getInstance()` then `new Logger` will throw an error

File: server/config/logger.js [hunk 4]
```diff
@@ -1,8 +1,56 @@
...
+  error: 5,
+};
+const getDateFormat = () => {
+  let date = new Date().toDateString();
+  let time = new Date().toLocaleTimeString();
+  return `${date} || ${time}`;
+};
 
+class Logger {
+  constructor(lvl) {
```
Comment (quinnlangille) [inline server/config/logger.js:29]: When we make a `new Logger` in `getInstance` we need to pass it the log level from ENV

**Pull request merged**